### PR TITLE
Rewrite implementation of container execution utils

### PR DESCRIPTION
**Pull request by osmman**

### CODE REVIEW

File: core/src/main/java/org/jboss/qa/jcontainer/util/executor/ProcessBuilderExecutor.java [hunk 1]
```diff
@@ -18,33 +18,38 @@
...
+	private ProcessBuilderExecutor() {
+	}
 
 	public static int syncExecute(final ProcessBuilder processBuilder) throws InterruptedException, IOException {
-		return syncExecute(processBuilder, null);
+		try {
+			return ProcessExecutor.builder().processBuilder(processBuilder).redirectError(true).build().syncExecute();
+		} catch (ExecutionException e) {
+			log.error(e.getMessage(), e);
+			return 600;
```
Comment (basovnik) [inline core/src/main/java/org/jboss/qa/jcontainer/util/executor/ProcessBuilderExecutor.java]: Use rather some constants.

File: core/src/main/java/org/jboss/qa/jcontainer/util/executor/ProcessExecutor.java [hunk 2]
```diff
@@ -0,0 +1,196 @@
...
+			}
+		}
+
+		@Override
+		public int waitFor() throws InterruptedException {
+			try {
+				return future.get();
+			} catch (ExecutionException e) {
+				log.error(e.getMessage(), e);
+				return 500;
```
Comment (basovnik) [inline core/src/main/java/org/jboss/qa/jcontainer/util/executor/ProcessExecutor.java]: Use constants.

File: core/src/main/java/org/jboss/qa/jcontainer/util/executor/ProcessExecutor.java [hunk 3]
```diff
@@ -0,0 +1,196 @@
...
+
+		final Process proxyProcess = new ProxyProcess(process, future);
+
+		executeService.shutdown();
+		return proxyProcess;
+	}
+
+	@Slf4j
+	@AllArgsConstructor
+	static class ProxyProcess extends Process {
```
Comment (basovnik) [inline core/src/main/java/org/jboss/qa/jcontainer/util/executor/ProcessExecutor.java]: What about to use `ProcessWrapper`?

File: containers/karaf/src/main/java/org/jboss/qa/jcontainer/karaf/KarafContainer.java [hunk 4]
```diff
@@ -85,6 +88,19 @@ public void run() {
 				}
 			}
 		}));
+		addShutdownHook(new Thread(new Runnable() {
```
Comment (basovnik) [inline containers/karaf/src/main/java/org/jboss/qa/jcontainer/karaf/KarafContainer.java]: This hook can be shared.

File: core/src/test/java/org/jboss/qa/jcontainer/util/executor/AbstractProcessBuilderExecutorTest.java [hunk 5]
```diff
@@ -0,0 +1,65 @@
...
+
+import lombok.extern.slf4j.Slf4j;
+
+@Slf4j
+public class AbstractProcessBuilderExecutorTest {
+
+	public static final String OUTPUT_TEXT = "output text";
+	public static final String ERROR_TEXT = "error text";
+
+	public static final String EXPECTED_RESULT = OUTPUT_TEXT + "\n" + ERROR_TEXT + "\n";
```
Comment (basovnik) [inline core/src/test/java/org/jboss/qa/jcontainer/util/executor/AbstractProcessBuilderExecutorTest.java]: `System.lineSeparator()`

File: core/src/test/java/org/jboss/qa/jcontainer/util/executor/LoopProcessTest.java [hunk 6]
```diff
@@ -0,0 +1,94 @@
...
+	@Test
+	public void loopProcessRedirectIntoFile() throws IOException, InterruptedException {
+		final File file = temporaryFolder.newFile();
+		final ProcessBuilder pb = new ProcessBuilder(commands);
+		final Process process = ProcessBuilderExecutor.asyncExecute(pb, file);
+		Thread.sleep(3000);
+		log.info("Terminate process");
+		process.destroy();
+		assertEquals(0, process.waitFor());
+		assertThat(FileUtils.readFileToString(file), allOf(containsString("Sleep over")));
```
Comment (basovnik) [inline core/src/test/java/org/jboss/qa/jcontainer/util/executor/LoopProcessTest.java]: Change `allOf` to `is`.

**Comment by osmman** [comment]
@basovnik I fixed all comments.

**Pull request merged**